rotkehlchen/tests/unit/test_ethereum_inquirer.py

```diff
@@ -206,6 +208,38 @@ def test_call_contract(ethereum_inquirer, ethereum_manager_connect_at_start):
     assert result >= 0
 
 
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_manager_connect_at_start', [(INFURA_ETH_NODE, ETHEREUM_ETHERSCAN_NODE)])  # noqa: E501
+def test_rpc_request_timeout(
+        ethereum_inquirer: 'EthereumInquirer',
+        ethereum_manager_connect_at_start: list[WeightedNode],
+) -> None:
+    """Test that rpc timeout errors result in the node being marked as `failed to connect`.
+    Regression test for https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=111711513
+    """
+    wait_until_all_nodes_connected(
+        connect_at_start=ethereum_manager_connect_at_start,
+        evm_inquirer=ethereum_inquirer,
+    )
+    yearn_ycrv_vault = ethereum_inquirer.contracts.contract(string_to_evm_address('0x5dbcF33D8c2E976c6b560249878e6F1491Bca25c'))  # noqa: E501
+
+    def mock_post(url, data, **kwargs):
+        data_j = json.loads(data)
+        if data_j.get('method') == 'eth_call':
+            raise requests.ReadTimeout
```

Comment:
Can you make two tests? One for read and one for connect? In the same pytest test, just also test for connection timeout.